common/src/Model/AttributableNode.cpp

```diff
@@ -672,5 +672,13 @@ namespace TrenchBroom {
             ensure(attributable != nullptr, "attributable is null");
             kdl::vec_erase(m_killTargets, attributable);
         }
+
+        bool operator==(const AttributableNode& lhs, const AttributableNode& rhs) {
```

Comment:
Suggest calling this `bool AttributableNode::equalIgnoringHierarchy(const AttributableNode& other)`. Or `equalContents` or `equalPayload`, because it's a particular notion of equality.

It's not comparing, from Node:
```
VisibilityState m_visibilityState;
LockState m_lockState;
```

Depending on the intended use case for these equality methods, you might want to include these. (e.g. for Layers we save/load these from the map now, so they're not just transient information.)

Reply:
I think that LayerNode should override == itself and check for that info if neccessary. For generic attributable nodes, I only want to consider the attributes.